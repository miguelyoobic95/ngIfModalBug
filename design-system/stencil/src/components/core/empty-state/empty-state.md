---
name: Empty State
category: Core Components
---

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`type`|string|   |the type of empty state|

```yoo-empty-state.html
    <yoo-empty-state></yoo-empty-state>
    <yoo-empty-state type="mission"></yoo-empty-state>
    <yoo-empty-state type="campaign"></yoo-empty-state>
    <yoo-empty-state type="location"></yoo-empty-state>
    <yoo-empty-state type="user"></yoo-empty-state>
    <yoo-empty-state type="note"></yoo-empty-state>
    <yoo-empty-state type="feed"></yoo-empty-state>
    <yoo-empty-state type="chat"></yoo-empty-state>
    <yoo-empty-state type="photo"></yoo-empty-state>
    <yoo-empty-state type="hq"></yoo-empty-state>
    <yoo-empty-state type="calendar"></yoo-empty-state>
    
```