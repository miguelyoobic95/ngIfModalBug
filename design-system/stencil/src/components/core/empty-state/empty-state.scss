
:host {
    display: block;
    width: 100%;
}

.container {
    margin: auto;
    text-align: center;
    padding: 1rem;
    img {
        max-height: 200px;
        width: 100%;
    }
}