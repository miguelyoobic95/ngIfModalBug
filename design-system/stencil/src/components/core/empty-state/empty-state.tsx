import { Component, Element, Prop } from '@stencil/core';
//import { setAnimation, animations } from '../../../utils/anim';

@Component({
    tag: 'yoo-empty-state',
    styleUrl: 'empty-state.scss',
    scoped: true
})
export class YooEmptyStateComponent {

    @Prop() type: string;

     @Element() host: HTMLStencilElement;

    componentDidLoad() {
        //setAnimation(animations.shake, this.host, { duration: 1000 });
    }

    getIconSrc() {
        return './assets/empty-states/' + (this.type || 'empty') + '.svg';
    }

    render() {
        return (
            <div class="container">
                <img src={this.getIconSrc()} />
            </div>
        );
    }
}
