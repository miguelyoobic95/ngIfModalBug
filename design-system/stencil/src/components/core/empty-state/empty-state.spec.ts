import { TestWindow } from '@stencil/core/testing';
import { YooEmptyStateComponent } from './empty-state';

jest.mock('../../../utils/anim');

describe('YooEmptyStateComponent', () => {
    let window;
    beforeEach(() => {
        window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooEmptyStateComponent()).toBeTruthy();
    });

    describe('rendering', () => {
        it ('should render', async () => {
            let element = await window.load({
                components: [YooEmptyStateComponent],
                html: '<yoo-empty-state></yoo-empty-state>'
            });
            expect(element).toMatchSnapshot();
        });
        it('should create an empty state of a given type', async () => {
            let element = await window.load({
                components: [YooEmptyStateComponent],
                html: '<yoo-empty-state type="type"></yoo-empty-state>'
            });
            expect(element).toMatchSnapshot();
        });
    });
    describe('helper methods', () => {
        it ('should return the correct path if no type is specified', async () => {
            let comp = new YooEmptyStateComponent();
            let assetPath = comp.getIconSrc();
            expect(assetPath).toEqual('./assets/empty-states/empty.svg');
        });
        it ('should return the correct path if the type is specified', async () => {
            let comp = new YooEmptyStateComponent();
            comp.type = 'type';
            let assetPath = comp.getIconSrc();
            expect(assetPath).toEqual('./assets/empty-states/type.svg');
        });
    });
});