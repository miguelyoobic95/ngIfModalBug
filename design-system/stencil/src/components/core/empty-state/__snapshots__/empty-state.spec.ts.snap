// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooEmptyStateComponent rendering should create an empty state of a given type 1`] = `
<yoo-empty-state
  class="hydrated"
  data-yoo-empty-state-host=""
  type="type"
>
  
  <div
    class="container"
    data-yoo-empty-state=""
  >
    <img
      data-yoo-empty-state=""
      src="./assets/empty-states/type.svg"
    />
  </div>
</yoo-empty-state>
`;

exports[`YooEmptyStateComponent rendering should render 1`] = `
<yoo-empty-state
  class="hydrated"
  data-yoo-empty-state-host=""
>
  
  <div
    class="container"
    data-yoo-empty-state=""
  >
    <img
      data-yoo-empty-state=""
      src="./assets/empty-states/empty.svg"
    />
  </div>
</yoo-empty-state>
`;
